src/main/java/seedu/address/model/group/UniqueGroupList.java

```diff
@@ -74,6 +74,21 @@ public void setGroups(List<Group> groups) {
         internalList.setAll(groups);
     }
 
+    public void setGroup(Group target, Group editedGroup) {
```

Comment:
The API could be `setGroup(int index, Group editedGroup)` instead since the caller already knows the index. This saves having to look up the target group again.

That said, since this is the same API that the PersonList uses, I think it's fine to keep it the same.